src/__init__.py

```diff
@@ -87,6 +87,25 @@ def root(netid):
 	return render_template("home.html", netid=netid)
 
 
+@app.context_processor
+def inject_header_values():
+	netid = auth.get_netid()
+	is_staff = netid and common.is_staff(netid)
+
+	def switcher_endpoint(course, assignment, mode="student"):
```

Comment:
Very clever solution! It'll be great if you can add some comment to this function to explain why it's needed and what's it's roughly doing, since it's not quite obvious from reading the code.

Reply:
Thanks, will do!